api/nodes/permissions.py

```diff
@@ -63,6 +64,28 @@ def has_object_permission(self, request, view, obj):
             has_auth = parent_node.can_edit(auth) and pointer_node.can_edit(auth)
             return has_auth
 
+
+class ContributorOrPublicForComments(permissions.BasePermission):
+
+    def has_object_permission(self, request, view, obj):
+        auth = get_user_auth(request)
+        if isinstance(obj, CommentReport):
+            comment_id = request.parser_context['kwargs']['comment_id']
+            comment = Comment.load(comment_id)
+            obj = comment.node
+        if isinstance(obj, Comment):
```

Comment:
This permissions method is also used by `NodeCommentsList` so the object needs to be an instance of `Node`, `Comment`, or `CommentReport`